<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:macro name="make_single_motor_ros2_control" params="name">

    <ros2_control name="${name}" type="system">
      <hardware>
        <plugin>unitree_hardware/UnitreeHardware</plugin>
        <param name="serial_interface">/dev/ttyUSB0</param>
      </hardware>
      <joint name="pan_joint">
        <param name="id">0</param>
        <param name="motor_type">GO-M8010-6</param>
        <param name="pos_gain">0.1</param>
        <param name="vel_gain">0.01</param>
        <param name="temperature_limit">80.</param>
        <command_interface name="position"/>
        <command_interface name="velocity"/>
        <command_interface name="effort"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
      </joint>
      <transmission name="pan_joint_transmission">
        <plugin>transmission_interface/SimpleTransmission</plugin>
        <actuator name="pan_motor" role="pan_motor"/>
        <joint name="pan_joint" role="pan_joint">
            <mechanical_reduction>1</mechanical_reduction>
        </joint>
      </transmission>
    </ros2_control>
  </xacro:macro>

</robot>